core/src/main/java/io/questdb/cairo/sql/Function.java

```diff
@@ -252,6 +266,49 @@ default boolean isUndefined() {
         return getType() == ColumnType.UNDEFINED;
     }
 
+    /**
+     * DeepClone does not clone the running-states, only the init states of the function,
+     * must be called before Function::init.
+     * Used for parallel filter/groupBy execution, so that avoid Function::parser for every worker.
+     * Note: CursorFunction and WindowFunction are not supported now.
+     *
+     * @return deep clones of the function
+     */
+    default Function deepClone() {
+        try {
+            Class<?> cls = this.getClass();
+            Constructor<?> funcCons = cls.getDeclaredConstructors()[0];
+            funcCons.setAccessible(true);
+            int parameterCount = funcCons.getParameterCount();
+            Object[] pArgs = new Object[parameterCount];
+            for (int i = 0; i < parameterCount; i++) {
+                pArgs[i] = DEFAULT_PRIMITIVE_VALUES.get(funcCons.getParameterTypes()[i]);
+            }
+            Object cloneFunc = funcCons.newInstance(pArgs);
+
+            while (cls != null) {
+                for (Field field : cls.getDeclaredFields()) {
+                    Class<?> fType = field.getType();
+                    field.setAccessible(true);
+                    if (field.get(this) == null || Modifier.isStatic(field.getModifiers()) || (field.get(cloneFunc) != null && !fType.isPrimitive())) {
```

Comment:
`Field#get()` has a cost, so we should try to do less calls: e.g. here we check `field.get(this) == null` and further `field.get(this)` can be done in one go with a local variable.